src/components/LogoutListener/helpers/useNativeAuthLogoutWarning/useNativeAuthLogoutWarning.ts

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useRef } from 'react';
+import BigNumber from 'bignumber.js';
+import { useGetAccountInfo } from 'hooks';
+import { useSelector } from 'reduxStore/DappProviderContext';
+import { loginInfoSelector } from 'reduxStore/selectors';
+import { getTokenExpiration } from 'services/nativeAuth/methods';
+import { addNewCustomToast, storage } from 'utils';
+import { faRefresh } from '@fortawesome/free-solid-svg-icons';
+import { localStorageKeys } from 'utils/storage/local';
+
+import { tokenExpierationToastWarningSeconds } from '../../constants';
+import { getHumanReadableTokenExpirationTime } from './helpers';
+
+export const useNativeAuthLogoutWarning = () => {
+  const { address } = useGetAccountInfo();
+  const { tokenLogin } = useSelector(loginInfoSelector);
+
+  const { secondsUntilExpires, expiresAt } = getTokenExpiration(
+    tokenLogin?.nativeAuthToken
+  );
+
+  const warningLogoutTimeoutRef = useRef<NodeJS.Timeout>();
+
+  useEffect(() => {
+    // Handle the actual logout functionality.
+    const secondsUntilExpiresBN = new BigNumber(String(secondsUntilExpires));
+    const millisecondsUntilLogout = secondsUntilExpiresBN.times(1000);
+
+    if (!secondsUntilExpires || secondsUntilExpiresBN.lte(0)) {
+      return;
+    }
+
+    // Handle the logout warning popup.
+    if (!storage.local.getItem(localStorageKeys.logoutWarningDismissed)) {
```

Comment:
I would not create a storage entry for this. If user refreshes the page, he should see the popup again (in my oppinion)

Reply:
Removed. 👍